src/main/navigation/navigation_multicopter.c

```diff
@@ -213,6 +215,14 @@ static void applyMulticopterAltitudeController(timeUs_t currentTimeUs)
         posControl.flags.verticalPositionDataConsumed = 1;
     }
 
+        if ( isThrustFacingDownwards()  & FlipInProcess ){  //KBI
```

Comment:
Altitude code should not be messing with setpoint itself. `applyMulticopterAltitudeController`'s task is **keeping** altitude, not changing the setpoint.

The code, however, should check for `isThrustFacingDownwards()` and keep `rcCommand[THROTTLE` at minimum throttle until the quad is upright again.

Reply:
Thank you very much for all of your time and help with these issues.  I have already tried adding a throttle constraint if 3d mode is enabled and the quad is inverted to keep the throttle low.  I also added a constraint if not inverted and 3d mode is enabled to keep the throttle positive.  This seemed to work but the quad was jumping around just after the flip and was losing altitude.  This is why I tried resetting the position at the top of the flip (should be the high point) to prevent further altitude loss and prevent the quad from erratically bouncing around trying to return to a previous position.  Should I go back to the throttle constraint approach or do you have some mock-up code modifications that I can test out.

Thanks again for your advise and all of the help you have been providing.

Reply:
@KenImhof I'll send you some code to test hopefully today. Btw please have a look at #2171 - since you are flying 3D mode you input would be very much appreciated.